PolyEngine/Engine/Src/DebugDrawComponents.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "ComponentBase.hpp"
+#include "DebugConfig.hpp"
+#include "Dynarray.hpp"
+#include "Vector.hpp"
+
+namespace Poly
+{
+	/// Component used for labeling entities for which the debug primitives 
+	/// will be drawn.
+	class ENGINE_DLLEXPORT DebugDrawableComponent : public ComponentBase
+	{
+	public:
+		DebugDrawableComponent(DebugDrawPreset preset) : entityPreset(preset) { };
+
+		const DebugDrawPreset entityPreset;
+	};
+
+	REGISTER_COMPONENT(ComponentsIDGroup, DebugDrawableComponent)
+
+	class ENGINE_DLLEXPORT DebugDrawLinesComponent : public ComponentBase
+	{
+	public:
+		DebugDrawLinesComponent()
+		{ }
+
+		struct DebugLine
+		{
+			Mesh::Vector3D Begin;
```

Comment:
I use this structure to create a vertex buffer for rendering. I'm not sure if I can pass data to GPU as array of `Vectors` (can I? so far I failed, idk why. another thing: `sizeof(Vector)` returns `32`, at least that's what IntelliSense says) and to convert whole array for buffering only to change type seems a bit over the top.

Reply:
No you can't, `Vector` is aligned differently (aligned to 16 bytes) due to using `__m128` variables. I didn't know this was passed directely to OpenGL. If so, we should move `Vector3D` out of `Mesh` structure because it's no longer part of `Mesh`. But still, we should minimize it's usage as much as possible. For example I don't want functions to return or accept `Vector3D`. Propper comments should be provided in it's implementation header. This is blocking the merg of this PR.